beacon_node/network/src/network_beacon_processor/mod.rs

```diff
@@ -775,6 +776,56 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                 "error" => %e)
         });
     }
+
+    fn publish_blobs_or_data_column(&self, blobs_or_data_column: BlobsOrDataColumns<T::EthSpec>) {
+        let messages = match blobs_or_data_column {
+            BlobsOrDataColumns::Blobs(blobs) => blobs
+                .into_iter()
+                .map(|blob| PubsubMessage::BlobSidecar(Box::new((blob.index, blob))))
+                .collect(),
+            BlobsOrDataColumns::DataColumns(columns) => columns
+                .into_iter()
+                .map(|column| {
+                    let subnet = DataColumnSubnetId::from_column_index::<T::EthSpec>(
+                        column.index as usize,
+                        &self.chain.spec,
+                    );
+                    PubsubMessage::DataColumnSidecar(Box::new((subnet, column)))
+                })
+                .collect(),
+        };
+        self.send_network_message(NetworkMessage::Publish { messages })
+    }
+
+    pub async fn fetch_engine_blobs_and_publish(
+        self: &Arc<Self>,
+        block: Arc<SignedBeaconBlock<T::EthSpec, FullPayload<T::EthSpec>>>,
+        block_root: Hash256,
+    ) {
+        let self_cloned = self.clone();
+        let log_cloned = self.log.clone();
+        let publish_fn = move |blobs_or_data_column| {
+            match &blobs_or_data_column {
+                BlobsOrDataColumns::Blobs(b) => {
+                    debug!(log_cloned, "Publishing blobs from EL"; "count" => b.len());
+                }
+                BlobsOrDataColumns::DataColumns(d) => {
+                    debug!(log_cloned, "Publishing data columns built from EL blobs"; "count" => d.len());
+                }
```

Comment:
You can move this log statements inside `publish_blobs_or_data_column` where `self.log` is already available

Reply:
Done